### DISCUSSION TITLE
Why can't we get pathName in server component ?

### DISCUSSION THREAD

**Discussion by anhVuducTimego** [vercel/next.js#58212] [Help] [answered]
### Summary

HI, I just want to highlight a menu link in a server component by compare pathname with href property in link, but look like there is no way to get pathname on server.
I must use client component to get pathname. 
I just wonder, Why create a client component for something small?

**Comment by mutasim77** [comment, marked as answer]
just use ```headers``` from next:
```js
import { headers } from "next/headers";

const header = headers();
const pathname = header.get('next-url')
console.log('pathName:', pathname);
```

**Reply by levchenkod** [reply]
This probably should not be marked as an answer, as usage of `next-url` is not recommended 

See https://github.com/vercel/next.js/issues/57762#issuecomment-1802052863

**Reply by kimhwanhoon** [reply]
But using header breaks the static build and it changes to dynamic..
